### PR TITLE
grub: Add config for using UART 2 on hikey

### PR DESCRIPTION
**Pull request by michpappas**
Signed-off-by: Michalis Pappas <mpappas@fastmail.fm>

### CODE REVIEW

**Comment by michpappas** [comment]
UART2 is a secondary console interface provided on the 40-Pin Low Speed Connector. The UART Serial mezzanine board allows choosing between the UART2 and the default UART3 using jumpers. Although UART3 is considered as the default interface, the Secure96 mezzanine board only uses UART2. This patch adds an additional configuration for grub to use UART2. Upon merge, I'll issue another PR with the remaining changes required in OP-TEE/build/

File: grub/grub_uart2.cfg [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+set default="0"
+set timeout=10
+
+menuentry 'GNU/Linux (eMMC) UART2' {
+    linux /Image console=tty0 console=ttyAMA2,115200 earlycon=pl011,0xf7112000 root=/dev/disk/by-partlabel/system rootwait rw ignore_loglevel efi=noruntime
+    initrd /initrd.img
+    devicetree /hi6220-hikey.dtb
+}
+
+menuentry 'GNU/Linux (eMMC) UART0' {
```
Comment (vchong) [inline grub/grub_uart2.cfg:12]: Any reason for UART0 here? If not, I'd rather this be the entry for UART3 since it's the default.

Comment (michpappas) [reply grub/grub_uart2.cfg:12]: It's for consistency with grub_uart3.cfg, which also provides UART0 as the alternative option.

The problem with having both UART2 and UART3 is that if you're able see grub's menu it means that you are using the correct console already, so there's probably not much point of being able to select between the ones on the expansion connector.

Still, if you'd rather to have them all in one file, I can send an update.

Comment (vchong) [reply grub/grub_uart2.cfg:12]: No, you're right. UART0 is better. Thanks!

**Review by vchong — APPROVED** [review]
`Reviewed-by: Victor Chong <victor.chong@linaro.org>`

**Comment by michpappas** [comment]
@vchong thanks, updated commit with your RB

**Comment by vchong** [comment]
Merged

**Pull request merged**